### PR TITLE
magento/magento2#: Replace deprecated functionality in “customer:hash:upgrade” command

### LINKED ISSUE #29611: [Issue] Replace deprecated functionality in “customer:hash:upgrade” command
This issue is automatically created based on existing pull request: magento/magento2#27333: magento/magento2#: Replace deprecated functionality in “customer:hash:upgrade” command [WIP]

---------

### Description (*)

PR replaces deprecated `customer->load()`, `customer->save()` functionality in the 

```bin/magento customer:hash:upgrade```

command.

Original PR: https://github.com/magento/magento2/pull/24243

CC: @dmytro-ch @ihor-sviziev 

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by atwixfirster**
### Description (*)

PR replaces deprecated `customer->load()`, `customer->save()` functionality in the 

```bin/magento customer:hash:upgrade```

command.

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/29611

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

Thank you!

### CODE REVIEW

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
@atwixfirster , could you please check the automated tests failures?

**Comment by atwixfirster** [comment]
@dmytro-ch
> @atwixfirster , could you please check the automated tests failures?

There are no any errors. Each test has been passed when you open a details page. 

Thanks

**Comment by atwixfirster** [comment]
@magento run all tests

**Comment by atwixfirster** [comment]
@dmytro-ch, I do not understand the reason why did tests fail. Do you have any ideas here?

Thank you!

**Comment by dmytro-ch** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch. Thank you for your request. I'm working on Magento instance for you

**Comment by dmytro-ch** [comment]
@atwixfirster, according to the Database Compare build failures, most likely the latest changes are breaking the Magento setup install process during the data upgrade step.
Could you please check this issue?

**Comment by atwixfirster** [comment]
> @atwixfirster, according to the Database Compare build failures, most likely the latest changes are breaking the Magento setup install process during the data upgrade step.
> Could you please check this issue?

@dmytro-ch, I've checked Magento installation through shell and Setup Wizards interfaces and I did not notice `bin/magento customer:hash:upgrade` call. Also, this command was not called within sample data installation. 

So, I am not sure that changes in the `Magento\Customer\Console\Command\UpgradeHashAlgorithmCommand` has been caused `Database Compare` build failures.

Thank you!

**Comment by sidolov** [comment]
@magento run all tests

File: app/code/Magento/Customer/Console/Command/UpgradeHashAlgorithmCommand.php [hunk 1]
```diff
@@ -65,19 +91,31 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+
+    /**
+     * Get customer data object
+     *
+     * @param int $customerId
+     * @return CustomerInterface
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    private function getCustomerDataObject(int $customerId)
```
Comment (sidolov) [inline app/code/Magento/Customer/Console/Command/UpgradeHashAlgorithmCommand.php]: What's the purpose of this method? It would be better inline it

**Review by sidolov — CHANGES_REQUESTED** [review]
@atwixfirster I have investigated the build log and found a fatal error on it: 
```
PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento2_prod.core_config_data' doesn't exist in /var/www/html/lib/internal/Magento/Framework/DB/Statement/Pdo/Mysql.php:91
```
Looks like something went wrong during Magento installation and tables in DB were not created. 
Could you please take a look?

File: app/code/Magento/Customer/Console/Command/UpgradeHashAlgorithmCommand.php (hunk 1 above)
Comment (atwixfirster) [reply app/code/Magento/Customer/Console/Command/UpgradeHashAlgorithmCommand.php]: > It would be better inline it

**implemented**

**Comment by atwixfirster** [comment]
> `PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento2_prod.core_config_data' doesn't exist in /var/www/html/lib/internal/Magento/Framework/DB/Statement/Pdo/Mysql.php:91`

@sidolov , I definitely do not understand what exactly in my fix can cause mentioned error. Do you have any suggestions how I can reproduce `'magento2_prod.core_config_data' doesn't exist` issue?

Thank you!

**Comment by sidolov** [comment]
@atwixfirster looks like something went wrong during the installation. Have you tried to install Magento with your fix?

**Comment by dmytro-ch** [comment]
Hi @atwixfirster,
I'm closing this PR due to inactivity.
Feel free to reopen the PR, once you have a chance to continue working on this one.
Thank you!

**Comment by atwixfirster** [comment]
> @atwixfirster looks like something went wrong during the installation. Have you tried to install Magento with your fix?

Yes, I've installed Magento with my fix. As I mentioned before Magento does not run `customer:hash:upgrade` command within installation process.

**Comment by atwixfirster** [comment]
@sidolov @dmytro-ch I've commented my changes and tests has failed.. So, I am not sure what's wrong.

Do you have any suggestions?

Thank you!

**Comment by dmytro-ch** [comment]
Closing this PR. We will proceed with further investigation in scope of #25538
Thank you!

**Pull request closed without merging**